### ISSUE TITLE
Support for cucumber 2 (rc's)

### ISSUE THREAD

**Issue by nicojs** [serenity-js/serenity-js#28]
I believe support for cucumber 2 is missing at the moment. Is that correct?

```bash
$ npm ls serenity-js
+-- UNMET PEER DEPENDENCY cucumber@2.0.0-rc.7
`-- serenity-js@1.2.0
```

Installed both cucumber@2.0.0-rc7 and serenity-js 1.2.0.

```bash
node ..\..\node_modules\protractor\bin\protractor 
(node:11616) DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[08:00:18] I/launcher - Running 1 instances of WebDriver
[08:00:18] I/direct - Using ChromeDriver directly...
[08:00:22] E/launcher - Error: TypeError: Cannot call a class as a function
Error: TypeError: Cannot call a class as a function
```

I believe that error has to do with the cli changes in cucumber. See [this PR for `protractor-cucumber-framework`](https://github.com/mattfritz/protractor-cucumber-framework/pull/66)

**Comment by jan-molak** [comment]
Hey @nicojs and thanks for giving Serenity/JS a try!

You're right, the project currently supports the latest stable version of Cucumber.js, which at the time of writing is 1.3.x.

To properly support Cucumber 1.3 and 2.x in a single code base I'll need to move Serenity/JS onto something like [Lerna](https://lernajs.io/) and [Lerna Semantic Release](https://github.com/atlassian/lerna-semantic-release), which I was planning to do before Cucumber 2.x becomes stable. Once Serenity/JS becomes a multi-module project, different test framework adapters could depend on different versions of the same package.

**Comment by nicojs** [comment]
Ok, thanks for the update.

**Comment by jan-molak** [comment]
Speaking of which, it would be great to know how many people are already using Cucumber 2.x?
If you'd like me to prioritise this functionality, please 👍 the OP's ticket.

**Comment by wswebcreation** [comment]
Using Cucumber 2 now in a project without Serenity, but if you need some help with it let me know.

**Comment by russoturisto** [comment]
Hello,

We are writing an Angular 5 project and are currently using vanilla Cucumber 3.1.  What is the status on the support for Cucumber 2+?

Thanks, :)

**Comment by vasiokos** [comment]
Is there any news regarding serenity and latest cucumber versions?
Thanks,

**Comment by jan-molak** [comment]
Yes, it's a WIP, I'm planning to work on that right after #162, which seems more pressing.

**Comment by gkushang** [comment]
@jan-molak when are you projecting to release the support for Cucumber@4?

**Comment by russoturisto** [comment]
Hello,

Any news on support for Cucumber 2+?

Thanks, :)

**Comment by matt-blanchette** [comment]
This would be nice to have support of Cucumber expressions for step definitions

**Comment by jan-molak** [comment]
My work on #162 ended up with me re-writing most of the internals of `@serenity-js/core`... I should be able to publish that within the next couple of weeks and then this ticket should be fairly easy to tackle as well. Stay tuned!

**Comment by wywrotowiec** [comment]
Hello,
I see that support for Cucumber 2+ was added on 2.0 branch. Is it stable enough for start using it ?Any news about official release ?

**Comment by jan-molak** [comment]
Hi @wywrotowiec! [Serenity/JS alpha](https://www.npmjs.com/package/@serenity-js/core/v/next) is available on npm right now.
You can reference it in your `package.json` as `@serenity-js/core: "next"`.

Now, here's the **small print**: I haven't written the developer guide for it yet, nor any nice configuration abstractions (that's my plan for the next few weeks), so while the main modules are stable and the APIs will most likely change very little if at all, the way you configure the framework will change.

If you'd like to use the alpha version now you'd need to rely on the [examples](https://github.com/jan-molak/serenity-js/tree/2.0/examples) and a gitter chat I'm setting up as well. Give me a shout if you're feeling adventurous and want to try the `alpha`, or if you'd rather wait for the official release (which I'm hoping to finish within the next few weeks).

**Comment by wywrotowiec** [comment]
Hi Jan!
 Thank you for your matter-of-fact response and sharing  info  about upcoming release that everyone is  excited about . We are just starting our adventure in js tests and for sure we will go with new version. If it is mater of few weeks we will wait for official release  to give better  first impression for developers and more chance of successful introduction.

**Comment by iranicus** [comment]
Currently sorting out a protractor-cucumber setup now for one of our angular 4 (to be updated to angular 8 this year) projects, to which I'll be using cucumber 5. The plan being for me to switch over to serenity 2 and pitch the framework likely mid 2020, taking a look at the commits recently to the serenity 2 alpha version I spotted the support for cucumber 5 referencing this ticket so looking forward to coming back to this later in the year to try it out :>. Thanks much.

**Comment by jan-molak** [comment]
Hi all! Serenity/JS 2.0 is live and ships with a brand new [Cucumber adapter](https://serenity-js.org/modules/cucumber/) that supports any version of Cucumber.js, all the way from 1.x to 6.x

Check out the [template project](https://github.com/serenity-js/serenity-js-cucumber-protractor-template) and [examples](https://github.com/serenity-js/serenity-js/tree/master/examples) and let me know what you think!
